### PR TITLE
WIP: Patch for #264 - build sRGB and Linux

### PR DESCRIPTION
**Pull request by manhnt9**

### CODE REVIEW

**Comment by manhnt9** [comment]
Finished, now Linux users can build with `cmake .. -DIRR_PCH=OFF`

**Comment by devshgraphicsprogramming** [comment]
ok but does PCH work on Linux or not?

File: source/Irrlicht/CMakeLists.txt [hunk 1]
```diff
@@ -292,6 +292,11 @@ add_library(IrrlichtServer STATIC
 	$<TARGET_OBJECTS:zlib>
 )
 
+if (NOT MSVC)
+	target_compile_options(Irrlicht PRIVATE -fexceptions)
```
Comment (devshgraphicsprogramming) [inline source/Irrlicht/CMakeLists.txt:296]: why enable exceptions?

Comment (manhnt9) [reply source/Irrlicht/CMakeLists.txt:296]: check other mails too, I explained it in the issue

File: 3rdparty/CMakeLists.txt [hunk 2]
```diff
@@ -18,9 +18,10 @@ add_subdirectory(glslang)
 set(SHADERC_SKIP_TESTS ON CACHE BOOL "Skip shaderc tests?" FORCE)
 set(SHADERC_SKIP_INSTALL ON CACHE BOOL "Install shaderc?" FORCE)
 # if it doesn't work without the `touch` on Linux, then fetch the latest submodule head of shaderc and try again
-#if (UNIX)
-#	file(WRITE ${THIRD_PARTY_SOURCE_DIR}/shaderc/libshaderc/libshaderc_combined.a "") #https://github.com/google/shaderc/issues/568
-#endif()
+# https://github.com/google/shaderc/issues/568
+if (UNIX)
+	file(WRITE ${THIRD_PARTY_SOURCE_DIR}/shaderc/libshaderc/libshaderc_combined.a "")
```
Comment (devshgraphicsprogramming) [inline 3rdparty/CMakeLists.txt:23]: I think this may fail if the output directory is not present there yet

File: 3rdparty/openssl/CMakeLists.txt [hunk 3]
```diff
@@ -1,22 +1,3 @@
-find_package(Git )
-
-if (GIT_FOUND)
```
Comment (devshgraphicsprogramming) [inline 3rdparty/openssl/CMakeLists.txt:3]: why did you remove this?

File: 3rdparty/CMakeLists.txt [hunk 4]
```diff
@@ -3,20 +3,20 @@ execute_process(COMMAND git submodule update --init --recursive
 )
 
 
-option(SPIRV_HEADERS_SKIP_EXAMPLES "SPIRV-Headers examples" ON)
+set(SPIRV_HEADERS_SKIP_EXAMPLES ON CACHE BOOL "Skip SPIRV-Headers examples" FORCE)
```
Comment (devshgraphicsprogramming) [inline 3rdparty/CMakeLists.txt]: A. Why are you using `FORCE` all over the place?
B. All of the subfolders specify the things with `option` not `set`, why are you using `set` ?

File: 3rdparty/CMakeLists.txt (hunk 2 above)
Comment (manhnt9) [reply 3rdparty/CMakeLists.txt:23]: The folder is guaranteed by the submodule (this is a verified fix by Criss and shaderc's devs). Still a dirty work-around though but it's out of my support scope anyways.

File: 3rdparty/openssl/CMakeLists.txt (hunk 3 above)
Comment (manhnt9) [reply 3rdparty/openssl/CMakeLists.txt:3]: explained in other issue

File: 3rdparty/CMakeLists.txt (hunk 4 above)
Comment (manhnt9) [reply 3rdparty/CMakeLists.txt]: here: https://github.com/buildaworldnet/IrrlichtBAW/issues/264

Comment (manhnt9) [reply 3rdparty/CMakeLists.txt]: Using `FORCE` so CMake overwrite current settings and people don't need to clear build folder and run CMake again. A minor issue though.

**Comment by manhnt9** [comment]
The PCH is not working, I disable PCH for now since I don't have firm experience with it.

**Comment by devshgraphicsprogramming** [comment]
>The PCH is not working [on Linux], I disable PCH for now since I don't have firm experience with it.

Ok that will be @Crisspl's task.

**Comment by devshgraphicsprogramming** [comment]
@manhnt9 come back with another PR for the exception throwing from `replace_invalid` in the utf8 dependency

**Comment by manhnt9** [comment]
Will keep an eye on that, hope the utf lib gets updates soon.
